### PR TITLE
Domains: Render feature gate instead of redirect for domain only sites

### PR DESCRIPTION
**Pull request by yurynix**
For domain only site, certain features are disabled
instead of redirecting we should render a page
that explains that.

This replaces #10840

screen shot 2017-02-01 at 14 11 45
  
#### Testing instructions
  
1. Run `git checkoutadd/fgate-domain-only-site` and start your server, or open a [live branch](https://calypso.live/?branch=add/fgate-domain-only-site)
2. Open some restricted page for a domain only site, for example click the 'Write Button' on the masterbar [`Post` page](http://calypso.localhost:3000/post/)
3. Check that you displayed with an empty content page that saying you need to upgrade as shown at the screenshot
  
#### Reviews
  
- [x] Code
- [x] Product

### CODE REVIEW

File: client/my-sites/controller.js [hunk 1]
```diff
@@ -113,6 +113,26 @@ function renderNoVisibleSites( context ) {
...
+
+	removeSidebar( reactContext );
+
+	renderWithReduxStore(
+		React.createElement( EmptyContentComponent, {
+			title: i18n.translate( 'This feature is not available for domains' ),
+			line: i18n.translate( 'To use this feature you need to create a site' ),
+			action: i18n.translate( 'Create New Site' ),
+			actionURL: '//dashboard.wordpress.com/wp-admin/index.php?page=my-blogs',
+			secondaryAction: i18n.translate( 'Manage Domain' ),
```
Comment (a8ci18n) [inline client/my-sites/controller.js]: Hi! I've found a possible matching string that has already been translated **28** times:
`translate( 'Domain Management' )` *ES Score: 7.50*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/fgate-domain-only-site)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/controller.js [hunk 2]
```diff
@@ -113,6 +113,26 @@ function renderNoVisibleSites( context ) {
...
+	const { store: reduxStore } = reactContext;
+
+	removeSidebar( reactContext );
+
+	renderWithReduxStore(
+		React.createElement( EmptyContentComponent, {
+			title: i18n.translate( 'This feature is not available for domains' ),
+			line: i18n.translate( 'To use this feature you need to create a site' ),
+			action: i18n.translate( 'Create New Site' ),
+			actionURL: '//dashboard.wordpress.com/wp-admin/index.php?page=my-blogs',
```
Comment (aidvu) [inline client/my-sites/controller.js]: Mmm, shouldn't we send to a Calypso flow instead of `wp-admin`?

File: client/my-sites/controller.js [hunk 3]
```diff
@@ -137,7 +157,7 @@ function onSelectedSiteAvailable( context ) {
 
 	if ( isDomainOnlySite( state, selectedSite.ID ) &&
 		! isPathAllowedForDomainOnlySite( context.pathname, selectedSite.slug ) ) {
```
Comment (aidvu) [inline client/my-sites/controller.js]: `isPathAllowedForDomainOnlySite` allows:
domains/manage/test.wordpress.com/email/test.wordpress.com
domains/manage/test.wordpress.com/email-forwarding/test.wordpress.com

Would be nice to check if we allow those, and to actually go through all of the paths, as we have transfer between blogs, and transfer between users that start with `/domains/manage`.

File: client/my-sites/controller.js (hunk 2 above)
Comment (drewblaisdell) [inline client/my-sites/controller.js]: We need to edit this once the flow exists in M2 anyway, so I think this is fine for now.

File: client/my-sites/controller.js (hunk 3 above)
Comment (drewblaisdell) [inline client/my-sites/controller.js]: I added a commit to change this to a canonical whitelist instead of a whitelist of prefixes.

**Comment by drewblaisdell** [comment]
If you're cool with my change, the code LGTM.

**Comment by yurynix** [comment]
Thank you @drewblaisdell and @aidvu !

**Pull request merged**